.github/workflows/build.yaml

```diff
@@ -18,47 +18,105 @@ jobs:
-    - name: Build and publish Dapr Flask Extension
-      if: github.event_name != 'pull_request'
-      run: |
-        cd ext/flask_dapr
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
-    - name: Build and publish dapr-ext-grpc
-      if: github.event_name != 'pull_request'
-      run: |
-        cd ext/dapr-ext-grpc
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
-    - name: Build and publish dapr-ext-fastapi
-      if: github.event_name != 'pull_request'
-      run: |
-        cd ext/dapr-ext-fastapi
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
+      - name: Set up Dapr CLI
+        run: wget -q ${{ env.DAPR_INSTALL_URL }} -O - | /bin/bash -s ${{ env.DAPR_CLI_VER }}
+      - name: Set up Go ${{ env.GOVER }}
+        if: env.DAPR_REF != '' || env.DAPR_CLI_REF != ''
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GOVER }}
+      - name: Checkout Dapr CLI repo to override dapr command.
+        uses: actions/checkout@v2
+        if: env.DAPR_CLI_REF != ''
+        with:
+          repository: dapr/cli
+          ref: ${{ env.DAPR_CLI_REF }}
+          path: cli
+      - name: Checkout Dapr repo to override daprd.
+        uses: actions/checkout@v2
+        if: env.DAPR_REF != ''
+        with:
+          repository: dapr/dapr
+          ref: ${{ env.DAPR_REF }}
+          path: dapr
+      - name: Build and override dapr cli with referenced commit.
+        if: env.DAPR_CLI_REF != ''
+        run: |
+          cd cli
+          make
+          sudo cp dist/linux_amd64/release/dapr /usr/local/bin/dapr
+          cd ..
+      - name: Initialize Dapr runtime ${{ env.DAPR_RUNTIME_VER }}
+        run: |
+          dapr uninstall --all
+          dapr init --runtime-version ${{ env.DAPR_RUNTIME_VER }}
+      - name: Build and override daprd with referenced commit.
+        if: env.DAPR_REF != ''
+        run: |
+          cd dapr
+          make
+          mkdir -p $HOME/.dapr/bin/
+          cp dist/linux_amd64/release/daprd $HOME/.dapr/bin/daprd
+          cd ..
+      - name: Override placement service.
+        if: env.DAPR_REF != ''
+        run: |
+          docker stop dapr_placement
+          cd dapr
+          ./dist/linux_amd64/release/placement --healthz-port 9091 &
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ env.PYTHON_VER }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_VER }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine tox
+      - name: Run Linter
+        run: |
+          tox -e flake8
+      - name: Check Typing
+        run: |
+          tox -e type
+      - name: Run unit-tests
+        run: |
+          tox -e py37
+      - name: Check Examples
+        run: |
+          tox -e examples
+      - name: Upload test coverage
```

Comment:
can be done before examples.

Reply:
> quick question should the validation be done in a separate workflow similar to quickstarts?

It could. I mostly did it this way for expediency, but it would probably be cleaner as a separate workflow. Let me split it out.